pkg/prowgen/prowgen_test.go

```diff
@@ -0,0 +1,77 @@
+package prowgen
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+type semVTest struct {
+	name     string
+	branches map[string]Branch
+	want     []string
+}
+
+func TestDeleteOldBranches(t *testing.T) {
+	tests := []semVTest{
+		{
+			name: "no previous minor, two previous majors",
+			branches: map[string]Branch{
+				"release-v3.1": {},
```

Comment:
Ideally we could only use the branch name only and pass an array instead of a map, but not that important I would say.